src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -800,15 +838,193 @@ def __try_add_error(error_list, detail):
+            self.__assessment_packages_copy = self.__assessment_substatus_msg_copy['patches']
+
+        if installation_substatus_index is not None:    # If installation data exists
+            self.__installation_substatus_msg_copy = self.__get_substatus_message(truncated_status_file, installation_substatus_index)
+            self.__installation_packages_copy = self.__installation_substatus_msg_copy['patches']
+            low_pri_index = self.__get_installation_low_pri_index(self.__installation_packages_copy)
+
+        status_file_without_package_list_size = self.size_of_constant_status_data(copy.deepcopy(truncated_status_file), assessment_substatus_index, installation_substatus_index)  # Deepcopy fully copy the object avoid reference modification
+        size_of_max_packages_allowed_in_status = self.__internal_file_capacity - status_file_without_package_list_size
+
+        while status_file_size_in_bytes > self.__internal_file_capacity:
+            # Start truncation process
+            packages_retained_in_assessment, packages_removed_from_assessment, packages_retained_in_installation, packages_removed_from_installation = \
+                self.__apply_truncation_process(self.__assessment_packages_copy, self.__installation_packages_copy, size_of_max_packages_allowed_in_status, low_pri_index)
+
+            if len(packages_removed_from_assessment) > 0:
+                # Recompose truncated status file payload (assessment)
+                truncated_status_file = self.__recompose_truncated_status_file(truncated_status_file=truncated_status_file, truncated_package_list=packages_retained_in_assessment,
+                    truncated_substatus_msg=self.__assessment_substatus_msg_copy, substatus_index=assessment_substatus_index)
+
+            if len(packages_removed_from_installation) > 0:
+                # Recompose truncated status file payload (installation)
+                truncated_status_file = self.__recompose_truncated_status_file(truncated_status_file=truncated_status_file, truncated_package_list=packages_retained_in_installation,
+                    truncated_substatus_msg=self.__installation_substatus_msg_copy, substatus_index=installation_substatus_index)
+
+            status_file_size_in_bytes = self.__calc_status_size_on_disk(json.dumps(truncated_status_file))
+            status_file_agent_size_diff = status_file_size_in_bytes - self.__internal_file_capacity
+            size_of_max_packages_allowed_in_status -= status_file_agent_size_diff   # Reduce the max packages byte size by tombstone, new error, and escape chars byte size
+
+        self.composite_logger.log_debug("End package list truncation")
+
+        return truncated_status_file
+
+    def __split_assessment_list(self, assessment_packages):
+        """ Split package list, keep 5 minimum packages, and remaining packages for truncation """
```

Comment:
""" Split assessment list into 2 parts. MIN_ASSESSMENT_PACKAGE_TO_RETAIN count as one part and remaining as the other """

This function is simply splitting the list, no need to reference the term 'truncation' in here. 

Similarly rename the variables used in this function to: min_assessment_patches_to_retain and remaining_assessment_patches

Reply:
done

Reply:
@rane-rajasi to resolve if satisfied.